core/src/Cuda/Kokkos_Cuda.hpp

```diff
@@ -129,13 +129,16 @@ class Cuda {
 
   /// \brief True if and only if this method is being called in a
   ///   thread-parallel function.
-  KOKKOS_INLINE_FUNCTION static int in_parallel() {
+
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  KOKKOS_DEPRECATED KOKKOS_INLINE_FUNCTION static int in_parallel() {
 #if defined(__CUDA_ARCH__)
     return true;
 #else
     return false;
 #endif
   }
+#endif
```

Comment:
I would request that the decision reasons are motivated, either in this PR or in an issue that this PR links to.

From an external viewpoint, this decision is hard to understand.

Reply:
See https://github.com/kokkos/kokkos/issues/6032#issuecomment-1662765779.